### DISCUSSION TITLE
Getting access to pending TX in ganache:vm:tx:{before, after, step} events?

### DISCUSSION THREAD

**Discussion by cd1m0** [ConsenSys-archive/ganache#4429] [Ideas]
Hi All

Is it possible to get access to the currently pending/executing transaction in the `"ganache:vm:tx:{before, after, step}"` event callbacks? 

Thanks
Best
~Dimo

**Comment by cd1m0** [comment]
In addition, could we add an arg telling if this `vm:tx` event comes from a eth_call, eth_sendTransaction, or debug_traceTransaction?

**Comment by davidmurdoch** [comment]
There is the `newPendingTransactions` event via the `eth_subscribe` RPC which will let you when a new transaction is added to the internal transaction pool. In programmatic use (where you instantiate ganache from JavaScript, not the command line tool) the `"ganache:vm:tx:{before, after, step}"` events will be emitted.

There is no mechanism to listen to these events via the RPC.

Adding information about the RPC method that initiated the event seems very reasonable. Can you open a new issue for that request?
